Helmet router material

rymaju opened the pull request:
Added Helmet, and some example usage of React Router DOM, Helmet, and Material UI.

A lot of stuff was left out to keep things vague enough for general use, but this should be a (decent???) starting point.

![image](https://user-images.githubusercontent.com/23691775/88000246-0c71de80-cab2-11ea-9fff-c046b2e4ff39.png)
![image](https://user-images.githubusercontent.com/23691775/88000256-14318300-cab2-11ea-853d-fdbd276dfc24.png)

rymaju committed:
installed helmet

rymaju committed:
Installed helmet types

rymaju committed:
Basic Examples

rymaju committed:
responsive fix

rymaju committed:
fix tabs warnings

rymaju commented:
Decided not to use react router sitemap, it would require moving the Router logic intro a separate jsx file. I think that thats a bit too messy from an organizational standpoint. We could just write our own sitemap.xml if we wanted to.

jackblanc commented on src/App.tsx:
```diff
@@ -1,26 +1,38 @@
-          Learn React
-        </a>
-      </header>
+    <div>
```
you could use a fragment here `<> {content} </>`

rymaju replied:
Not super sure what you mean here, probably some lack of React knowledge on my part. I'll look it up and try to fix it. Let me know if I get it wrong

jackblanc commented on src/App.tsx:
```diff
@@ -1,26 +1,38 @@
+      <Helmet>
+        <meta
+          name='keywords'
+          content='C4C,code,for,community,code4community,codeforcommunity,northeastern,boston'
```
how do spaces affect this?

rymaju replied:
The delimiter is just the commas, so adding spaces wouldn't make a difference. I can add spaces to make it prettier if needed.

jackblanc commented on src/components/ExamplePage.tsx:
```diff
@@ -0,0 +1,58 @@
+  CardActions,
+  Button
+} from '@material-ui/core'
+function ExamplePage () {
```
I generally prefer `const ExamplePage = () => {` as opposed to this syntax, though I'm not sure that there's any technical difference

rymaju replied:
Thats probably better for "this" scoping, I'll make those changes.

jackblanc commented on src/components/ExamplePage.tsx:
```diff
@@ -0,0 +1,58 @@
+  CardContent,
+  CardActions,
+  Button
+} from '@material-ui/core'
```
There should be a space here, which would be something caught by tslint, is that set up in this repo? I didn't do anything to do so, curious if it just works out of box

rymaju replied:
I didn't have the tslint extension when I was coding this so I didn't have any linting, I'll check everything over again!

jackblanc commented on src/components/ExamplePage.tsx:
```diff
@@ -0,0 +1,58 @@
+          Hey, this is example text size body2.
+        </Typography>
+
+        <br />
```
Can we use something other than `<br />`? I feel like that's hard to work with long term

rymaju replied:
The spacing was just so you can properly see the shadow of the paper/cards which are usually used in a grid with automatic spacing. I can wrap it in a Box with some extra y margin instead.

jackblanc commented on src/components/Home.tsx:
```diff
@@ -0,0 +1,76 @@
+          </Typography>
+
+          <Typography variant='subtitle2' gutterBottom>
+            Using React+Typescript and Material UI. Its probably better than
```
`It's`

also, remove that sentence 😂

jackblanc commented on src/components/Home.tsx:
```diff
@@ -0,0 +1,76 @@
+          <Grid item sm>
+            <Typography variant='h6'>Subheading1</Typography>
+            <Typography variant='body1'>
+              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
```
⛏️  (this means nitpick, not a huge deal)
If you could have different lorem ipsum for each that might look better :)

jackblanc commented on src/components/NotFound.tsx:
```diff
@@ -0,0 +1,31 @@
+  Container,
+  Box,
+  Typography,
+
```
def need some linting setup for stuff like this extra line

jackblanc reviewed (commented):
Hey sorry I was doing coop all day. This is a great start! If you could configure linting as a part of this that would be great but not absolutely necessary

rymaju committed:
fixes

rymaju commented:
- Made components in the form: `const Component = () => { ... }`
- Changed up the lorem ipsum
- Removed snarky comment
- Added spaces properly (still not sure if tslint is set up correctly but supposedly im linting on save)
- removed `<br/>`tags and replaced them with a <Box> wrapper with margin/padding

jackblanc commented on src/App.tsx:
```diff
@@ -8,9 +8,9 @@ import NotFound from './components/NotFound'
 import { AppBar, Tabs, Tab } from '@material-ui/core'
 
-function App () {
+const App = () => {
```
Should these components be typed? 

`const App: React.FC<> = () => {}`

jackblanc reviewed (approved):
Great work! One comment on typing components

rymaju committed:
Added type definitions for functional components
